### ISSUE TITLE
create configuration-check utility

### ISSUE THREAD

**Issue by ghost** [openSUSE/docserv#91]
when `docserv` is started, it does not check its own configuration/setup. that tends to lead to setup issues occurring. it would be nice to have a tool that checks for the following:

* [ ] check validity of the INI file (both bare INI syntax and that there are no attributes that docserv can't interpret)
* [ ] our mail setup works properly: while testing locally, docserv2 tried to send out mails even though my mail setup was not working (because my weirdo local hostname does not really work as the domain name behind an `@` char).
* [ ] ssh works with all servers (ideally allow checking for both `root` and `docserv` users)
* [ ] the XML config is stitchable
* [ ] check that the template is sane: there is a `template\.(html|php)`, a `res/`dir, a `translation.xml` that validates according to what we create for #87
* [ ] all directories referenced in the ini exist
* [ ] `docserv` user/group exist

**Comment by tomschr** [comment]
By accident, I came across an example of the Wiki article [systemd service file (as a sample see mariadb package)](https://en.opensuse.org/openSUSE:Packaging_guidelines#systemd_services_.28as_a_sample_see_mariadb_package.29):

```ini
[Unit]
Description=MySQL server
Wants=basic.target
Conflicts=mysql.target
After=basic.target network.target 

[Service]
Restart=on-abort
Type=simple
ExecStartPre=/usr/lib/mysql/mysql-systemd-helper  install
ExecStartPre=/usr/lib/mysql/mysql-systemd-helper  upgrade
ExecStart=/usr/lib/mysql/mysql-systemd-helper     start
ExecStartPost=/usr/lib/mysql/mysql-systemd-helper wait

[Install]
WantedBy=multi-user.target
```

Don't know if this is feasible, but if we use `ExecStartPre` and `ExecStartPost` for our own scripts, we could perhaps check the installation.
Not sure if we could implement all of your above wishes, but maybe a partial implementation could also help.

**Comment by sknorr** [comment]
well, i think it wouldn't hurt to have a sanity check in the service in addition ... but really, we still need the script that checks for all these things. and you're right, we could not check everything when starting the daemon:

* ssh might turn interactive, so the script would never exit
* testing the mail setup this way would take very long

**Comment by tomschr** [comment]
Yes, I agree. I assume we can have both. What do you think about the following idea?

First we create a script that does all the nitty-gritty. With subcommands, we can distinguish between "simple" and "hard" sanity checks. That allows us to activate only some. In the systemd service file, we can call the "simple" parts (as shown in the above listing).

I think, the advantage is, that we have to write only *one* script which can be used in different scenarios: as an admin when he wants to check something special and inside the system service file as a pre sanity check.

Would that work for you?

**Comment by tomschr** [comment]
@sknorr 

> check that the template is sane

what do you mean with "sane" here? Well-formed?

**Comment by svenseeberg** [comment]
Checking for a valid mail configuration is hard. I really see no way how to achieve this. And checking the XML configuration can already be checked by the stitch command.

The other items seem valid.

**Comment by sknorr** [comment]
checking mail functionality would presumably consist of the user entering an e-mail address, and then being asked whether they received a mail... granted, there is not a lot of troubleshooting help we can offer there but better than nothing imo.
